workflow-ui/backstack-android/src/androidTest/java/com/squareup/workflow1/ui/backstack/test/BackStackContainerTest.kt

```diff
@@ -0,0 +1,175 @@
+      activity.currentTestView.viewState = "hello world"
+    }
+
+    // Destroy and recreate the activity.
+    scenario.recreate()
+
+    // Check that the view state was actually restored.
+    assertThat(viewForScreen("initial").viewState).isEqualTo("hello world")
+  }
+
+  @Test fun restores_view_on_pop_after_config_change() {
+    lateinit var firstScreen: TestRendering
+
+    scenario.onActivity { activity ->
+      // Set some view state to be saved and restored.
+      activity.currentTestView.viewState = "hello world"
+      firstScreen = activity.backstack!!.top
+    }
+
+    // Navigate to another screen.
+    setBackstack(firstScreen, TestRendering("new screen"))
+    assertThat(viewForScreen("new screen").viewState).isEqualTo("")
+
+    // Destroy and recreate the activity.
+    scenario.recreate()
+
+    // Navigate back.
+    setBackstack(firstScreen)
+
+    // Check that the view state was actually restored.
+    assertThat(viewForScreen("initial").viewState).isEqualTo("hello world")
+  }
+
+  @Test fun state_rendering_and_attach_ordering() {
+    val events = mutableListOf<String>()
+    fun log(name: String, event: String, view: ViewStateTestView) {
+      events += "$name $event viewState=${view.viewState}"
+    }
+
+    fun consumeEvents() = events.toList().also { events.clear() }
+
+    fun testRendering(name: String) = TestRendering(
+      name = name,
+      onViewCreated = { view -> log(name, "onViewCreated", view) },
+      onShowRendering = { view -> log(name, "onShowRendering", view) },
+      viewHooks = object : ViewHooks {
+        override fun onSaveInstanceState(view: ViewStateTestView) = log(name, "onSave", view)
+        override fun onRestoreInstanceState(view: ViewStateTestView) = log(name, "onRestore", view)
+        override fun onAttach(view: ViewStateTestView) = log(name, "onAttach", view)
+        override fun onDetach(view: ViewStateTestView) = log(name, "onDetach", view)
+      }
+    )
+
+    val firstRendering = testRendering("first")
+    val secondRendering = testRendering("second")
+
+    // Setup initial screen.
+    setBackstack(firstRendering)
+    waitForScreen(firstRendering.name)
+    scenario.onActivity {
+      it.currentTestView.viewState = "hello"
+    }
+    assertThat(consumeEvents()).containsExactly(
+      "first onViewCreated viewState=",
+      "first onShowRendering viewState=",
+      "first onShowRendering viewState=",
+      "first onAttach viewState="
+    )
+
+    // Navigate forward.
+    setBackstack(firstRendering, secondRendering)
+    waitForScreen(secondRendering.name)
+    assertThat(consumeEvents()).containsExactly(
+      "second onViewCreated viewState=",
+      "second onShowRendering viewState=",
+      "second onShowRendering viewState=",
+      "first onSave viewState=hello",
+      "first onDetach viewState=hello",
+      "second onAttach viewState=",
+      "first onAttach viewState=hello",
+      "first onDetach viewState=hello"
```

Comment:
I don't know why the exiting view gets attached and detached at the end of the transition here.